modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java

```diff
@@ -1005,8 +1005,14 @@ public CPDefinition deleteCPDefinition(CPDefinition cpDefinition)
 
 		// Commerce product display layouts
 
-		cpDisplayLayoutLocalService.deleteCPDisplayLayouts(
-			CPDefinition.class, cpDefinition.getCPDefinitionId());
+		List<CPDisplayLayout> cpDisplayLayouts =
+			cpDisplayLayoutPersistence.findByC_C(
+				classNameLocalService.getClassNameId(CPDefinition.class),
+				cpDefinition.getCPDefinitionId());
+
+		for (CPDisplayLayout cpDisplayLayout : cpDisplayLayouts) {
+			cpDisplayLayoutPersistence.remove(cpDisplayLayout);
```

Comment:
@slnn The original code calls `CPDisplayLayoutLocalService.deleteCPDisplayLayout(CPDisplayLayout)` which has the `@Indexable` annotation. When you switch it to directly use the persistence, the index will not be updated.

```	@Indexable(type = IndexableType.DELETE)
	@Override
	public CPDisplayLayout deleteCPDisplayLayout(
		CPDisplayLayout cpDisplayLayout) {

		return cpDisplayLayoutPersistence.remove(cpDisplayLayout);
	}
```

Reply:
@slnn You may need to manually remove the indexes of `CPDisplayLayout` or find another way to break the circular dependency.

Reply:
cc @tinatian

Reply:
Hi @dantewang 

I add the index process manually in the last commit after persistence's remove method.

Lily